flink-libraries/flink-table/src/test/scala/org/apache/flink/table/TableSourceTest.scala

```diff
@@ -126,21 +156,49 @@ class TableSourceTest extends TableTestBase {
 
   @Test
   def testStreamProjectableSourceScanNoIdentityCalc(): Unit = {
-    val (csvTable, tableName) = tableSource
+    val (tableSource, tableName) = csvTable
     val util = streamTestUtil()
     val tEnv = util.tEnv
 
-    tEnv.registerTableSource(tableName, csvTable)
+    tEnv.registerTableSource(tableName, tableSource)
 
     val result = tEnv
       .scan(tableName)
       .select('id, 'score, 'first)
 
-    val expected = sourceStreamTableNode(tableName, noCalcFields)
+    val expected = projectableSourceStreamTableNode(tableName, noCalcFields)
     util.verifyTable(result, expected)
   }
 
   @Test
+  def testStreamFilterableSourceScanPlanTableApi(): Unit = {
+    val (tableSource, tableName) = filterableTableSource
+    val util = streamTestUtil()
+    val tEnv = util.tEnv
+
+    tEnv.registerTableSource(tableName, tableSource)
+
+    val result = tEnv
+      .scan(tableName)
+      .select('price, 'id, 'amount)
+      .where("amount > 2 && price * 2 < 32")
```

Comment:
An example for an unsupported predicate would be `'id.cast(BasicTypeInfo.STRING_TYPE_INFO) === "abc"`. This throws and exception when translating it to an `Expression`.

As said before, unsupported expressions should be gracefully handled by not failing but instead by not offering this `RexNode` to the `FilterableTableSource` and evaluating it in the `DataSetCalc`.
I would suggest to use `CAST` as an example to implement the graceful handling and adding support for it once the failure-free translation works.